src/Storage/Device/S3.php

```diff
@@ -107,9 +184,30 @@ public function move(string $source, string $target):bool
      *
      * @return bool
      */
+
+     /* 
+     DELETE /my-second-image.jpg HTTP/1.1
+    Host: <bucket>.s3.<Region>.amazonaws.com
+    Date: Wed, 12 Oct 2009 17:50:00 GMT
+    Authorization: authorization string
+    Content-Type: text/plain    
+     */
+
     public function delete(string $path, bool $recursive = false):bool
     {
-        return false;
+        $this->__resetResponse();
+        $verb = 'DELETE';
+        $uri = $path;
+        $uri = $uri !== '' ? '/'.str_replace('%2F', '/', rawurlencode($uri)) : '/';
```

Comment:
Seems like some part of handling the request are repetitive, should we abstract it with a protected `call` method?